src/common/include/inline_boundary_conditions.fpp

```diff
@@ -0,0 +1,280 @@
+                            q_prim_vf(i)%sf(k, n, l)
+                    end if
+                end do
+            end do
+        #:endif
+    #:elif DIR == "z"
+        #:if LOC == "L"
+            do i = 1, sys_size
+                do j = 1, buff_size
+                    if (i == momxb) then
+                        q_prim_vf(i)%sf(k, l, -j) = &
+                            -q_prim_vf(i)%sf(k, l, j - 1) + 2._wp*bc_z%vb1
+                    elseif (i == momxb + 1 .and. num_dims > 1) then
+                        q_prim_vf(i)%sf(k, l, -j) = &
+                            -q_prim_vf(i)%sf(k, l, j - 1) + 2._wp*bc_z%vb2
+                    elseif (i == momxb + 2 .and. num_dims > 2) then
+                        q_prim_vf(i)%sf(k, l, -j) = &
+                            -q_prim_vf(i)%sf(k, l, j - 1) + 2._wp*bc_z%vb3
+                    else
+                        q_prim_vf(i)%sf(k, l, -j) = &
+                            q_prim_vf(i)%sf(k, l, 0)
+                    end if
+                end do
+            end do
+        #:elif LOC == "R"
+            do i = 1, sys_size
+                do j = 1, buff_size
+                    if (i == momxb) then
+                        q_prim_vf(i)%sf(k, l, p + j) = &
+                            -q_prim_vf(i)%sf(k, l, p - (j - 1)) + 2._wp*bc_z%ve1
+                    elseif (i == momxb + 1 .and. num_dims > 1) then
+                        q_prim_vf(i)%sf(k, l, p + j) = &
+                            -q_prim_vf(i)%sf(k, l, p - (j - 1)) + 2._wp*bc_z%ve2
+                    elseif (i == momxb + 2 .and. num_dims > 2) then
+                        q_prim_vf(i)%sf(k, l, p + j) = &
+                            -q_prim_vf(i)%sf(k, l, p - (j - 1)) + 2._wp*bc_z%ve3
+                    else
+                        q_prim_vf(i)%sf(k, l, p + j) = &
+                            q_prim_vf(i)%sf(k, l, p)
+                    end if
+                end do
+            end do
+        #:endif
+    #:endif
+#:enddef
+
+#:def PRIM_DIRICHLET_BC(DIR, LOC, DEST, SRC)
+    do i = 1, sys_size
+        do j = 1, buff_size
+            q_prim_vf(i)%sf(${DEST}$) = bc_buffers(${DIR}$, ${LOC}$)%sf(${SRC}$)
+        end do
+    end do
+#:enddef
+
+#:def QBMM_BC(DEST, SRC)
+    do i = 1, nb
+        do q = 1, nnode
+            do j = 1, buff_size
+                pb(${DEST}$) = pb(${SRC}$)
+                mv(${DEST}$) = mv(${SRC}$)
+            end do
+        end do
+    end do
+#:enddef
+
+#:def COLOR_FUNC_SLIP_WALL_BC(DIR, DEST, SRC)
+    do i = 1, num_dims + 1
+        do j = 1, buff_size
+            if (i == ${DIR}$) then
+                c_divs(i)%sf(${DEST}$) = -c_divs(i)%sf(${SRC}$)
+            else
+                c_divs(i)%sf(${DEST}$) = c_divs(i)%sf(${SRC}$)
+            end if
+        end do
+    end do
+#:enddef
+
+#:def COLOR_FUNC_BC(DEST, SRC)
+    do i = 1, num_dims + 1
+        do j = 1, buff_size
+            c_divs(i)%sf(${DEST}$) = c_divs(i)%sf(${SRC}$)
+        end do
+    end do
+#:enddef
```

Comment:
I know I initially named this, but it should be renamed to `COLOR_FUNC_EXTRAPOLATION` for clarity.